lib/graphql/schema/field/scope_extension.rb

```diff
@@ -10,7 +10,13 @@ def after_resolve(object:, arguments:, context:, value:, memo:)
           else
             ret_type = @field.type.unwrap
             if ret_type.respond_to?(:scope_items)
-              ret_type.scope_items(value, context)
+              scoped_items = ret_type.scope_items(value, context)
+              if !scoped_items.equal?(value) && !ret_type.reauthorize_scoped_objects
```

Comment:
Per my comment, perhaps this should also check the type of value, so that it doesn't skip authorizing arrays twice?